fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/pattern/Pattern.java

```diff
@@ -71,4 +72,22 @@ public boolean matchWithoutChild(GroupExpression expression) {
 
         return getOpType().equals(expression.getOp().getOpType());
     }
+
+    public boolean matchWithoutChild(OptExpression expression) {
+        if (expression == null) {
+            return false;
+        }
+
+        // special for MergeLimitRule, avoid false when merge limit with scan
```

Comment:
Is the comment here right？

Reply:
Right, but unnecessary, I will remove it_